### PR TITLE
Feature/sav 1

### PR DESCRIPTION
**Pull request by avsharapov**
Сделал модельки для таблиц.
Немного реструктуризовал приложение для удобства. Добавил yaml файл с настройками.
Подключил swagger 2 для документирвоания rest api. Требуется его дальнейшая настройка.
Сделал тестовое заполнение БД данными для проверки работоспособности моделек и рестов.

Для работы нужна postgresql, настройки подключения к бд нужно указать в файле application.yml/

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -33,15 +33,25 @@ val test by tasks.getting(Test::class) {
...
     compile("org.jetbrains.kotlin:kotlin-reflect")
-    compile("com.fasterxml.jackson.module:jackson-module-kotlin")
+    compile("com.fasterxml.jackson.module:jackson-module-kotlin:2.9.4.1")
+    // swagger
+    compile("io.springfox:springfox-swagger2:2.8.0")
+    compile("io.springfox:springfox-swagger-ui:2.8.0")
+    // exposed
+    compile("org.jetbrains.exposed:exposed:0.10.2")
+    compile("org.jetbrains.exposed:spring-transaction:0.10.2")
+    runtime("org.postgresql:postgresql")
```
Comment (leonid133) [inline build.gradle.kts:54]: why did we agree to use mysql, and did you use postgres?

File: src/main/kotlin/nejen/AppProperties.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        var url: String = ""
+        var driver: String = ""
+        var username: String = ""
+        var password: String = ""
+    }
+
+    class Server {
+        var port: String = ""
+    }
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/AppProperties.kt:30]: new line at the end of file

File: src/main/kotlin/nejen/Application.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    companion object {
+        @JvmStatic
+        fun main(args: Array<String>) {
+            val context = SpringApplication.run(Application::class.java, *args)
+            val bean = context.getBean(AppProperties::class.java)
+            Database.connect(bean.database.url, driver = bean.database.driver, user = bean.database.username, password = bean.database.password)
+            TransactionManager.manager.defaultIsolationLevel = Connection.TRANSACTION_SERIALIZABLE
+            transaction {
+                logger.addLogger(StdOutSqlLogger)
+                SchemaUtils.drop(TenantsTable, RentsTable, ActTable, ShelfTable, OrdersTable, CashTable)
```
Comment (leonid133) [inline src/main/kotlin/nejen/Application.kt]: Drop the tables in main and filling in with test data this a bad practice. Need abstraction for that.

File: src/test/kotlin/controllers/SearchServiceTests.kt [hunk 4]
```diff
@@ -1,30 +1,29 @@
...
-        val content = """all tenantry"""
-        assertEquals(content, restTemplate.getForObject<String>("/tenantry"))
+        /*assertTrue(restTemplate.getForObject<List<TenantsDTO>>("/tenantry")!!.isNotEmpty())*/
     }
 
     @Test
     fun findByQuery(){
-        val content = """query"""
-        assertEquals(content, restTemplate.getForObject<String>("/tenantry?query=${content}"))
+        /*assertTrue(restTemplate.getForObject<String>("/tenantry?query=content")!!.isNotEmpty())*/
```
Comment (leonid133) [inline src/test/kotlin/controllers/SearchServiceTests.kt:26]: ?

**Review by leonid133 — CHANGES_REQUESTED** [review]

File: build.gradle.kts [hunk 5]
```diff
@@ -33,15 +33,25 @@ val test by tasks.getting(Test::class) {
...
     compile("org.springframework.boot:spring-boot-starter-web")
     compile("org.springframework.boot:spring-boot-starter-data-jpa")
+    compile("org.springframework.boot:spring-boot-starter-actuator")
+    compile("org.springframework.boot:spring-boot-configuration-processor")
     compile("com.h2database:h2")
     compile("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
     compile("org.jetbrains.kotlin:kotlin-reflect")
-    compile("com.fasterxml.jackson.module:jackson-module-kotlin")
+    compile("com.fasterxml.jackson.module:jackson-module-kotlin:2.9.4.1")
+    // swagger
```
Comment (leonid133) [inline build.gradle.kts:48]: Why?. this not apply to this task.

File: src/main/kotlin/nejen/Application.kt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                    orders = order
+                    sell = "3"
+                    take = "2"
+                    discount = "1"
+                }
+                commit()
+            }
+        }
+    }
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/Application.kt]: new line at the end of file

File: src/main/kotlin/nejen/controllers/SwaggerConf.kt [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+@Configuration
+@EnableSwagger2
+class SwaggerConf {
+    @Bean
+    fun api() : Docket = Docket(DocumentationType.SWAGGER_2)
+            .select()
+            .apis(RequestHandlerSelectors.any())
+            .paths(PathSelectors.any())
+            .build()
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/controllers/SwaggerConf.kt:20]: new line at the end of file

File: src/main/kotlin/nejen/models/Act.kt [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+fun ActItem.getDto(): ActDTO {
+    val actItem = this
+    return transaction {
+        return@transaction ActDTO(actItem.rents.getDto(), actItem.shelf.getDto(), actItem.start_date, actItem.stop_date, actItem.term, actItem.payment)
+    }
+}
+
+fun List<ActItem>.getDto(): List<ActDTO> {
+    return this.map { it.getDto() }
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/models/Act.kt:41]: new line at the end of file

File: src/main/kotlin/nejen/models/Cash.kt [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+fun CashItem.getDto(): CashDTO {
+    val cashItem = this
+    return transaction {
+        return@transaction CashDTO(cashItem.cash_date, cashItem.orders.getDto(), cashItem.sell, cashItem.take, cashItem.discount)
+    }
+}
+
+fun List<CashItem>.getDto(): List<CashDTO> {
+    return this.map { it.getDto() }
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/models/Cash.kt:39]: new line at the end of file

File: src/main/kotlin/nejen/models/Orders.kt [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+fun OrdersItem.getDto(): OrdersDTO {
+    val ordersItem = this
+    return transaction {
+        return@transaction OrdersDTO(ordersItem.act.getDto(), ordersItem.name_item, ordersItem.desc_item, ordersItem.qty, ordersItem.price)
+    }
+}
+
+fun List<OrdersItem>.getDto(): List<OrdersDTO> {
+    return this.map { it.getDto() }
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/models/Orders.kt:39]: new line at the end of file

File: src/main/kotlin/nejen/models/Rents.kt [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+fun RentsItem.getDto(): RentsDTO {
+    val rentsItem = this
+    return transaction {
+        return@transaction RentsDTO(rentsItem.number, rentsItem.start_date, rentsItem.stop_date, rentsItem.tenants.getDto())
+    }
+}
+
+fun List<RentsItem>.getDto(): List<RentsDTO> {
+    return this.map { it.getDto() }
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/models/Rents.kt:37]: new line at the end of file

File: src/main/kotlin/nejen/models/Shelf.kt [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+fun ShelfItem.getDto(): ShelfDTO {
+    val shelfItem = this
+    return transaction {
+        return@transaction ShelfDTO(shelfItem.name, shelfItem.price)
+    }
+}
+
+fun List<ShelfItem>.getDto(): List<ShelfDTO> {
+    return this.map { it.getDto() }
+}
```
Comment (leonid133) [inline src/main/kotlin/nejen/models/Shelf.kt:32]: new line at the end of file

**Review by leonid133 — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (avsharapov) [reply build.gradle.kts:54]: Firstly, technically, I do not see a difference between the two. And locally I have already deployed postgres. Secondly, h2 was initially connected here, not mySql.
Bottom line: when we want mySql, we just connect the driver and change the connection string to the database in the application.yml file.

Comment (avsharapov) [reply build.gradle.kts:54]: I do not see a problem here.

File: src/main/kotlin/nejen/AppProperties.kt (hunk 2 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/AppProperties.kt:30]: "new line at the end of file" what does it mean?

File: src/main/kotlin/nejen/Application.kt (hunk 3 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/Application.kt]: Yes, it's not here for long. Someday we'll just connect to the database. 
I guess I'll just delete it.

File: build.gradle.kts (hunk 5 above)
Comment (avsharapov) [reply build.gradle.kts:48]: The version is indicated in order that there should not be any questions and which one we use, and what documentation to look at. Specifically, this module has certain versions for certain versions of jackson.
And why do we need it at all?

File: src/test/kotlin/controllers/SearchServiceTests.kt (hunk 4 above)
Comment (avsharapov) [reply src/test/kotlin/controllers/SearchServiceTests.kt:26]: I have not yet figured out how to block the access to the database here.
And so I commented on an unnecessary and incorrect test. Is that the question?

File: src/main/kotlin/nejen/controllers/SwaggerConf.kt (hunk 7 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/controllers/SwaggerConf.kt:20]: What for?

File: src/main/kotlin/nejen/models/Act.kt (hunk 8 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/models/Act.kt:41]: What for?

File: src/main/kotlin/nejen/models/Cash.kt (hunk 9 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/models/Cash.kt:39]: What for?

File: src/main/kotlin/nejen/models/Orders.kt (hunk 10 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/models/Orders.kt:39]: What for?

File: src/main/kotlin/nejen/models/Rents.kt (hunk 11 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/models/Rents.kt:37]: What for?

File: src/main/kotlin/nejen/models/Shelf.kt (hunk 12 above)
Comment (avsharapov) [reply src/main/kotlin/nejen/models/Shelf.kt:32]: What for?

**Pull request merged**